src/compiler/checker.ts

```diff
@@ -18595,7 +18595,19 @@ namespace ts {
             // Grammar checking
             checkGrammarStatementInAmbientContext(node);
 
-            checkExpression(node.expression);
+            const type = checkExpression(node.expression);

```

Comment:
This example isn't "missed" in the same way - we don't know that the intended use isn't (*cough*)
```ts
await await bar();
```
This rule is just intended to catch places where a `Promise` is generated, then "evaporates" due to the value not be attached to any reference. It's the same reason
```ts
var x = someAsyncFunc();
```
isn't flagged as an error -- the next line might be `await x;`, for example